tests/user_input_checks_test.py

```diff
@@ -435,3 +436,47 @@ def test_passing_custom_density_estimator(arg):
         density_estimator = arg
     prior = MultivariateNormal(torch.zeros(2), torch.eye(2))
     _ = SNPE_C(prior=prior, density_estimator=density_estimator)
+
+
+def test_validate_theta_and_x_cpu():
+
+    cpu_device = torch.device('cpu')
+
+    theta = torch.ones((32,8), dtype=torch.float32).to(cpu_device)
+    x = torch.zeros((32,100), dtype=torch.float32).to(cpu_device)
+    otheta = torch.FloatTensor((32,8)) #using an explicit type
+
+    # Check assumptions in implementation:
+    # A cpu based torch.tensor is an instance of torch.FloatTensor,
+    # both of torch.FloatTensor and torch.tensor expose dtype==float32.
+    assert isinstance(theta, torch.Tensor)
+    assert theta.dtype == torch.float32
+    assert isinstance(theta, torch.FloatTensor)
+    assert otheta.dtype == torch.float32
+
+    # test on cpu
+    validate_theta_and_x(theta, x)
+
+    with pytest.raises(AssertionError) as exc:
+        validate_theta_and_x(theta, x.to(torch.float64))
+
+
+@pytest.mark.gpu
+def test_validate_theta_and_x_gpu():
+
+    gpu_if_present = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+
+    theta = torch.ones((32,8), dtype=torch.float32).to(gpu_if_present)
+    x = torch.zeros((32,100), dtype=torch.float32).to(gpu_if_present)
+
+    # A gpu based torch.tensor is NOT an instance of torch.FloatTensor,
+    # but rather of torch.cuda.FloatTensor, still
+    # tensor.dtype returns torch.float32.
+    assert isinstance(theta, torch.Tensor)
+    assert theta.dtype == torch.float32
+    assert not isinstance(theta, torch.FloatTensor)
```

Comment:
```suggestion
    assert not isinstance(theta, torch.FloatTensor), "When on the GPU, theta must not be a torch.FloatTensor."
```

Reply:
would it make sense / be more informative to `assert isinstance(theta, torch.cuda.FloatTensor)` then?

Reply:
did my best in 90f0f24 to resolve this